File: Java/SETPractice/src/main/java/SOLIDPrinciple/OpenClose/ShapeFactory.java
package SOLIDPrinciple.OpenClose;

import java.io.InvalidObjectException;

public class ShapeFactory {

    public IShape GetShapeInstance(Shape shape) throws InvalidObjectException {
        switch (shape) {
            case Circle:
                return new Circle();
            case Rectangle:
                return new Rectangle();
            default:
                throw new InvalidObjectException("Enter a valid shape");
        }
    }
}

File: DotNet/SETPractice/SOLIDPrinciples/OpenClose/ShapeFactory.cs
namespace SOLIDPrinciples.OpenClose
{
    public class ShapeFactory
    {
        public IShape GetShapeInstance(Shape shape)
        {
            switch (shape) 
            {
                case Shape.Rectangle:
                    return new Rectangle();
                case Shape.Circle:
                    return new Circle();
                default:
                    throw new InvalidOperationException("Please provide a valid shape");
            }
        }
    }
}
